code/modules/cargo/randomstock.dm

```diff
@@ -0,0 +1,1880 @@
+			//Improved parts, less common
+			"/obj/item/weapon/stock_parts/capacitor/adv" = 1,
+			"/obj/item/weapon/stock_parts/scanning_module/adv" = 1,
+			"/obj/item/weapon/stock_parts/manipulator/nano" = 1,
+			"/obj/item/weapon/stock_parts/micro_laser/high" = 1,
+			"/obj/item/weapon/stock_parts/matter_bin/adv" = 1,
+			//Top level parts, rare
+			"/obj/item/weapon/stock_parts/capacitor/super" = 0.3,
+			"/obj/item/weapon/stock_parts/scanning_module/phasic" = 0.3,
+			"/obj/item/weapon/stock_parts/manipulator/pico" = 0.3,
+			"/obj/item/weapon/stock_parts/micro_laser/ultra" = 0.3,
+			"/obj/item/weapon/stock_parts/matter_bin/super" = 0.3,
+			//Telecomms parts, useless novelties and red herrings.
+			"/obj/item/weapon/stock_parts/subspace/ansible" = 0.5,
+			"/obj/item/weapon/stock_parts/subspace/filter" = 0.5,
+			"/obj/item/weapon/stock_parts/subspace/amplifier" = 0.5,
+			"/obj/item/weapon/stock_parts/subspace/treatment" = 0.5,
+			"/obj/item/weapon/stock_parts/subspace/analyzer" = 0.5,
+			"/obj/item/weapon/stock_parts/subspace/crystal" = 0.5,
+			"/obj/item/weapon/stock_parts/subspace/transmitter" = 0.5
+			)
+
+			var/number = rand(2,5)
+			while (number > 0)
+				var/part = pickweight(parts)
+				new part(L)
+				number--
+
+
+		if ("cane")
+			if (prob(5))
+				new /obj/item/weapon/cane/concealed(L)
+			else if (prob(20))
+				new /obj/item/weapon/staff/broom(L)
+			else if (prob(30))
+				new /obj/item/weapon/staff/stick(L)
+			else
+				new /obj/item/weapon/cane(L)
+
+		if ("warning")
+			if (prob(50))
+				new /obj/item/weapon/caution(L)
+			else
+				new /obj/item/weapon/caution/cone(L)
+
+		if ("gasmask")
+			var/list/masks = list(
+			"/obj/item/clothing/mask/gas" = 10,
+			"/obj/item/clothing/mask/gas/plaguedoctor" = 1,
+			"/obj/item/clothing/mask/gas/swat" = 5,
+			"/obj/item/clothing/mask/gas/clown_hat" = 0.5,
+			"/obj/item/clothing/mask/gas/sexyclown" = 0.5,
+			"/obj/item/clothing/mask/gas/mime" = 0.5,
+			"/obj/item/clothing/mask/gas/monkeymask" = 0.5,
+			"/obj/item/clothing/mask/gas/sexymime" = 0.5,
+			"/obj/item/clothing/mask/gas/death_commando" = 2,
+			"/obj/item/clothing/mask/gas/cyborg" = 1,
+			"/obj/item/clothing/mask/gas/owl_mask" = 1
+			)
+
+			var/type = pickweight(masks)
+			new type(L)
+
+		if ("cleanernades")
+			new /obj/item/weapon/grenade/chem_grenade/cleaner(L)
+			new /obj/item/weapon/grenade/chem_grenade/cleaner(L)
+			if (prob(50))
+				new /obj/item/weapon/grenade/chem_grenade/cleaner(L)
+				new /obj/item/weapon/grenade/chem_grenade/cleaner(L)
+
+
+
+		if ("mining")
+			if (prob(50))
+				new /obj/item/weapon/shovel(L)
+			if (prob(50))
+				new /obj/item/weapon/pickaxe(L)
+			if (prob(50))
+				new /obj/item/clothing/glasses/material(L)
+			if (prob(50))
+				new /obj/item/device/flashlight/lantern(L)
+			if (prob(50))
+				new /obj/item/weapon/mining_scanner(L)
+
+		if ("paicard")
+			new /obj/item/device/paicard(L)
+
+		if ("phoronsheets")
+			new /obj/item/stack/material/phoron(L, rand(5,50))
+
+
```

Comment:
The comment block should be enough imho.
But feel free to leave it in if you want